### PR TITLE
DHFPROD-7892: Add endpoint to save Hub Central Config

### PR DESCRIPTION
**Pull request by ryanjdew**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by wooldridge — CHANGES_REQUESTED** [review]
The GET and PUT work great as described.

However, we need to be able to update properties for (one or more) entities in the config, or update global settings like scale, and keep any existing config data as-is. Currently everything gets overwritten with whatever we PUT in the payload. So, we need to merge the PUT JSON with whatever JSON already exists. I've added notes about this in the Jira:

https://project.marklogic.com/jira/browse/DHFPROD-7892

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java [hunk 1]
```diff
@@ -46,6 +52,50 @@
 @Controller
 @RequestMapping("/api/models")
 public class ModelController extends BaseController {
+    ObjectMapper objectMapper = new ObjectMapper();
+    @RequestMapping(value = "/hubCentralConfig", method = RequestMethod.GET)
+    @ResponseBody
+    @ApiOperation("This retrieves the Hub Central Config for models")
+    public ResponseEntity<?> getHubCentralConfig() throws FailedRequestException, ResourceNotFoundException, ForbiddenUserException {
+        return ResponseEntity.ok(getHubCentralConfigObject());
```
Comment (aniamy) [inline marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java:60]: Shouldn't we return an empty config if there is no configuration yet?

Comment (ryanjdew) [reply marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java:61]: Yes. The latest push handles it that way.

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java [hunk 2]
```diff
@@ -47,6 +53,27 @@
...
+    @ApiOperation("This retrieves the Hub Central Config for models")
+    public ResponseEntity<?> getHubCentralConfig() throws FailedRequestException, ResourceNotFoundException, ForbiddenUserException {
+        JacksonHandle handle = new JacksonHandle();
+        getHubClient().getFinalClient().newJSONDocumentManager().read("/config/hubCentral.json", handle);
+        return ResponseEntity.ok(handle.get());
+    }
+
+    @RequestMapping(value = "/hubCentralConfig", method = RequestMethod.PUT)
+    @ResponseBody
+    @ApiOperation("This retrieves the Hub Central Config for models")
```
[the remark this answers is not in the document]
Comment (ryanjdew) [reply marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java]: done

**Comment by ryanjdew** [comment]
@wooldridge I've updated the PR to handle the scenarios you outlined.

**Review by wooldridge — CHANGES_REQUESTED** [review]
@ryanjdew thanks for the update! Everything looks great, except I see there's no way to delete the config now once it has been set? See the end below, PUTting "{}" does nothing. I see this would be problematic for testing as we'd want to wipe the config value during the After All. Can we add a DELETE endpoint or am I missing something?

1. With reference-entity-model, do login POST and GET hubConfig: {}
2. PUT the following:
```
{
  "modeling": {
    "entities": {
      "BabyRegistry": {
        "color": "#EEEFF1",
        "graphX": 70,
        "graphY": 2,
        "icon": "someIcon"
      },
      "Customer": {
        "color": "#FF0000",
        "graphX": 120,
        "graphY": -20,
        "icon": "someOtherIcon"
      }
    },
    "scale": 1.2
  }
}
```
3. GET returns the above.
4. PUT the following:
```
{
  "modeling": {
    "entities": {
      "BabyRegistry": {
        "graphX": 88,
        "graphY": -44
      }
    }
  }
}
```
5. GET returns the following:
```
{
    "modeling": {
        "entities": {
            "BabyRegistry": {
                "color": "#EEEFF1",
                "graphX": 88,
                "graphY": -44,
                "icon": "someIcon"
            },
            "Customer": {
                "color": "#FF0000",
                "graphX": 120,
                "graphY": -20,
                "icon": "someOtherIcon"
            }
        },
        "scale": 1.2
    }
}
```
6. PUT the following:
```
{
  "modeling": {
    "scale": 0.5
  }
}
```
7. GET returns the following:
```
{
    "modeling": {
        "entities": {
            "BabyRegistry": {
                "color": "#EEEFF1",
                "graphX": 88,
                "graphY": -44,
                "icon": "someIcon"
            },
            "Customer": {
                "color": "#FF0000",
                "graphX": 120,
                "graphY": -20,
                "icon": "someOtherIcon"
            }
        },
        "scale": 0.5
    }
}
```
8. PUT the following:
```
{
  "modeling": {
    "entities": {
      "BabyRegistry": {
        "graphX": -1,
        "graphY": 20
      },
      "Customer": {
        "graphX": 123,
        "graphY": 456
      }
    }
  }
}
```
9. GET returns the following:
```
{
    "modeling": {
        "entities": {
            "BabyRegistry": {
                "color": "#EEEFF1",
                "graphX": -1,
                "graphY": 20,
                "icon": "someIcon"
            },
            "Customer": {
                "color": "#FF0000",
                "graphX": 123,
                "graphY": 456,
                "icon": "someOtherIcon"
            }
        },
        "scale": 0.5
    }
}
```
10. PUT the following:
```
{
  "modeling": {
    "entities": {
      "Product": {
        "graphX": 0,
        "graphY": 0
      }
    },
    "scale": 0
  }
}
```
11. GET returns the following:
```
{
    "modeling": {
        "entities": {
            "BabyRegistry": {
                "color": "#EEEFF1",
                "graphX": -1,
                "graphY": 20,
                "icon": "someIcon"
            },
            "Customer": {
                "color": "#FF0000",
                "graphX": 123,
                "graphY": 456,
                "icon": "someOtherIcon"
            },
            "Product": {
                "graphX": 0,
                "graphY": 0
            }
        },
        "scale": 0
    }
}
```
12. PUT the following:
```
{}
```
13. GET returns the following:
```
{
    "modeling": {
        "entities": {
            "BabyRegistry": {
                "color": "#EEEFF1",
                "graphX": -1,
                "graphY": 20,
                "icon": "someIcon"
            },
            "Customer": {
                "color": "#FF0000",
                "graphX": 123,
                "graphY": 456,
                "icon": "someOtherIcon"
            },
            "Product": {
                "graphX": 0,
                "graphY": 0
            }
        },
        "scale": 0
    }
}
```

**Comment by ryanjdew** [comment]
@wooldridge I've added a DELETE endpoint for clearing out the settings.

**Review by wooldridge — DISMISSED** [review]
DELETE now works, functionality looks good to me.

**Review by xnikhil08 — DISMISSED** [review]

**Review by AkshayBajajML — DISMISSED** [review]

**Review by aniamy — APPROVED** [review]

**Review by xnikhil08 — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Review by wooldridge — APPROVED** [review]
Ran my Postman tests on the current branch, everything looks OK with those manual tests.

**Pull request merged**